src/Interpreters/Context.h

```diff
@@ -351,6 +351,11 @@ class Context: public std::enable_shared_from_this<Context>
                                                         /// It's shared with all children contexts.
     MergeTreeTransactionHolder merge_tree_transaction_holder;   /// It will rollback or commit transaction on Context destruction.
 
+    /// For ON CLUSTER queries we check access on initiator node before adding query to the queue
+    /// so we don't need to check it again on queue processing.
+    /// Moreover, it may be not possible to check access on queue processing because we execute query internally, without a user.
```

Comment:
Maybe it's better to check if it's a secondary query in interpreters? This variable looks quite bad.

Reply:
Or we can check in some interpreters whether it's the global context and don't do any checks for access if so.